### ISSUE TITLE
sporadic failure on thread_spinlock/test_async.event_unset_from_handler/0

### ISSUE THREAD

**Issue by dmitrygladkov** [openucx/ucx#3522]
```
16:30:37 [ RUN      ] thread_spinlock/test_async.event_unset_from_handler/0
16:30:38 /scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node-upstream/worker/3/contrib/../test/gtest/common/test.cc:267: Failure
16:30:38 Failed
16:30:38 Got 2 warnings during the test
16:30:38 
[  FAILED  ] thread_spinlock/test_async.event_unset_from_handler/0, where GetParam() = 1 (229 ms)
```

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9987/label=hpc-test-node-upstream,worker=3/consoleFull

**Comment by brminich** [comment]
Also happened here:
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10318/label=hpc-test-node,worker=1/consoleFull#-853622487b816df86-9afc-46eb-984f-1a9a125d73bb
```
18:37:20 [ RUN      ] thread_spinlock/test_async.event_unset_from_handler/0
18:37:20 /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node/worker/1/contrib/../test/gtest/ucs/test_async.cc:643: Failure
18:37:20 Value of: le.count()
18:37:20   Actual: 0
18:37:20 Expected: 1
18:37:20 /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node/worker/1/contrib/../test/gtest/ucs/test_async.cc:647: Failure
18:37:20 Value of: le.count()
18:37:20   Actual: 0
18:37:20 Expected: 1
```

**Comment by yosefe** [comment]
Fixed by #3754
